### PR TITLE
refactor sample scripts

### PR DESCRIPTION
**Pull request by kosovan**
… removed some examples of bad practice (not all of them yet)

### CODE REVIEW

**Comment by pm-blanco** [comment]
Solves #92, and partially #94 
- `samples/peptide.py`: Removed completely legacy logic, running over different pH values. Now it stores the time series for a single input pH value
- `samples/analysis.py`: Added a new sample script to post-process time series.
- `testsuite/samples_tests.py`: Added a new functional test that triggers `samples/peptide.py` and `samples/analysis.py`.
- `lib/analysis.py`: Added a new functionality in `analyze_time_series`, an optional argument that allows ignoring filenames for the analysis.

**Comment by pm-blanco** [comment]
Solves #92 and #94 

### Changed
- The sample script `plot_HH.py` has been replaced for specific examples on how to plot data post-processed with pyMBE: `plot_branched_polyampholyte.py`, `plot_peptide.py`, and `plot_peptide_mixture_grxmc_ideal.py`. 
- Sample scripts now take the pH as an argparse input instead of looping over several pH values. This enables paralization of the sample scripts and avoids conflicts with the current post-processing pipeline. 

### Added
- New sample script showing how to use the analysis tools in pyMBE for post-processing time series from the sample scripts `analyze_time_series.py` (#95) 
- A new optional argument `ignore_files`  for `lib.analysis.analyze_time_series`, enabling to provide a list of files to be ignored for post-processing of time series. (#95)
- Functional testing for all sampling scripts. (#95)

IMO, this PR is now ready to be merged @kosovan. However, I am no longer qualified to be the reviewer of this PR because I have done a significant portion of the code so I will ask someone else from our dev. team to co-review it.

**Comment by pm-blanco** [comment]
@Zitzeronion if you have the time to take a look into this PR, I would appreciate your comments regarding if the current pipeline works and it is clear to you. I also wrote a README file in the samples folder explaining how to use the sample scripts.

**Comment by Zitzeronion** [comment]
Hi @pm-blanco  and @kosovan, I don't know if I'm the right person to give qualified feedback (I'm a fluid dynamics person with little knowledge on polymers, proteins and constant pH simulations), but I had a look at the changes. I overall like the new samples folder. The split into plotting and computing scripts make sense and run time arguments are a good choice. It is however maybe worth to write a section in the documentation on how use these new scripts (e.g. first compute with script X then plot with script Y).

There are just three things maybe worth changing:

- In [samples/README.md](samples/README.md) the reference is a footnote while in [samples/Beyer2024/README.md](samples/Beyer2024/README.md) it is not. 
- In [lib/analysis.py](lib/analysis.py) inside the `get_dt()` function, it is maybe worth to raise an error if `time[0] == time[1]`. I don't know if that will ever happen, but it would prevent dividing by 0.
- Last point is more a question, what is the difference between [samples/peptide.py] and [samples/Beyer2024/peptide.py], is it the inclusion of the hydrodynamic radius in the latter? Is it that you need both scripts as one serves as a sample and the other one is referenced in the publication?

I haven't run the scripts but could do that tomorrow if I should.

**Comment by pm-blanco** [comment]
@kosovan @davidbbeyer could you please finish the review of this PR by this week? We need to continue to make progress with the other issues in the library :smile:

**Comment by davidbbeyer** [comment]
@pm-blanco Yes, I was waiting for the review by @kosovan, but the two-week time frame we agreed on will be done by tomorrow. Then I will take care myself.

**Review by davidbbeyer — COMMENTED** [review]
Looks good to me apart from some small points. I ran all the tests and it works.

File: samples/plot_peptide_cpH.py
Comment (davidbbeyer) [inline samples/plot_peptide_cpH.py:37]: The default folder name should be ´peptide' instead of ´peptide_cpH' to be consistent with the production script

File: testsuite/cph_ideal_tests.py
Comment (pm-blanco) [reply testsuite/cph_ideal_tests.py]: Agreed, I just forgot to remove the legacy code that I kept there while refactoring the test :smile:

File: README.md
Comment (pm-blanco) [reply README.md]: Good idea, I added the link!

File: samples/plot_peptide_cpH.py
Comment (pm-blanco) [reply samples/plot_peptide_cpH.py:37]: Actually, I decided to solve this issue the other way around by consistently naming all folders "peptide_cpH" to clearly link it to the script that produced the data.

**Review by davidbbeyer — APPROVED** [review]
Looks good!

**Pull request merged**
